marlowe-dashboard-client/src/Template/View.purs

```diff
@@ -14,122 +14,130 @@ import Data.Map (toUnfoldable) as Map
+          [ a
+              [ classNames [ "flex", "items-center", "p-2", "font-semibold" ]
+              , onClick_ ToggleTemplateLibraryCard
+              ]
+              [ Icon.previous
+              , span_
+                  [ text "Choose template" ]
+              ]
+          ]
+      , contractNicknameDisplay contractName contractNickname editingNickname
       , div
-          [ classNames [ "absolute", "bottom-1", "left-1", "right-1", "flex", "items-center", "justify-between" ] ]
-          $ contractNavigationButtons setupScreen roleWallets wallets
+          [ classNames [ "h-full", "overflow-y-auto", "grid" ] ]
+          [ subHeader "top-0" Icon.roles "Roles"
+          , contractRoles wallets metaData extendedContract roleWallets
+          , subHeader "top-10" Icon.terms "Terms"
+          , contractParameters metaData templateContent
+          , subHeader "top-20" Icon.pay "Review and pay"
+          , contractReview state
+          ]
       ]
 
-contractSetupScreenHeader :: forall p. Screen -> String -> HTML p Action
-contractSetupScreenHeader setupScreen contractNickname =
-  div_
+contractNicknameDisplay :: forall p. String -> String -> Boolean -> HTML p Action
+contractNicknameDisplay contractName contractNickname editingNickname =
+  div
+    [ classNames [ "ml-12", "border-l", "border-darkgray", "p-2", "pl-6" ] ]
     [ div
-        []
-        [ input
-            [ classNames [ "w-full", "text-center" ]
-            , type_ InputText
-            , placeholder "Contract name"
-            , value contractNickname
-            , onValueInput_ SetContractNickname
-            ]
-        ]
-    , div
-        [ classNames [ "flex", "justify-between" ] ]
-        [ span
-            [ classNames $ screenClasses setupScreen ContractRolesScreen ]
-            [ text "Roles" ]
-        , span
-            [ classNames $ screenClasses setupScreen ContractParametersScreen ]
-            [ text "Parameters" ]
-        , span
-            [ classNames $ screenClasses setupScreen ContractReviewScreen ]
-            [ text "Review" ]
+        [ classNames [] ]
+        [ div
+            [ classNames [ "uppercase", "text-sm" ] ]
+            [ text contractName ]
+        , if editingNickname then
+            input
+              [ classNames [ "bg-transparent", "font-semibold" ]
+              , type_ InputText
+              , placeholder "Contract name"
+              , value contractNickname
+              , onValueInput_ SetContractNickname
+              , onBlur $ const $ Just ToggleEditingNickname
+              ]
+          else
+            div_
+              [ span
+                  [ classNames [ "font-semibold", "mr-2" ] ]
+                  [ text contractNickname ]
+              , a
+                  [ classNames [ "inline-block", "-mt-4", "text-sm" ]
+                  , onClick_ ToggleEditingNickname -- TODO: focus the nickname input
```

Comment:
you are probably going to require having an Halogen Ref to locate the input to focus. I found out of it rather recently 😅